src/tap.cpp

```diff
@@ -41,27 +48,135 @@ TAP::TAP(QObject *parent) : QObject(parent) {
+        exit(1);
+    }
+
+    close(sock);
+
+    QDBusConnection bus = QDBusConnection::systemBus();
+    QDBusInterface manager("net.connman", "/", "net.connman.Manager", bus);
+    QDBusMessage result = manager.call("GetServices");
+
+    const QDBusArgument argument = result.arguments().at(0).value<QDBusArgument>();
+    QMap<QDBusObjectPath, QVariantMap> reply;
+    argument >> reply;
+
+    for (auto it = reply.keyValueBegin(); it != reply.keyValueEnd(); ++it) {
+        const auto &key = it->first;
+        const auto &value = it->second;
+
+        if (value.contains("Ethernet")) {
+            //QString name = value["Ethernet"].toMap()["Name"].toString();
+            QDBusArgument ethernetArg = value["Ethernet"].value<QDBusArgument>();
+            QVariantMap ethernetMap;
+            ethernetArg >> ethernetMap;
+
+            const auto name = ethernetMap["Interface"].toString();
+
+            if (!strcmp(mIfaceName.toLocal8Bit().data(), name.toLocal8Bit().data())) {
+                qDebug() << "Will reset the IPv4 configuration";
+
+                QDBusInterface service("net.connman", key.path(), "net.connman.Service", bus);
+
+                {
+                    QDBusArgument propertiesArg = value["IPv4.Configuration"].value<QDBusArgument>();
+                    QVariantMap propertiesMap;
+                    propertiesArg >> propertiesMap;
+
+                    propertiesMap["Method"] = "manual";
+                    propertiesMap["Address"] = "10.0.2.3";
+                    propertiesMap["Netmask"] = "255.255.255.0";
+                    propertiesMap["Gateway"] = "10.0.2.2";
+
+                    QVariantList args;
+                    args << "IPv4.Configuration" << QVariant::fromValue(QDBusVariant(propertiesMap));
+                    service.callWithArgumentList(QDBus::AutoDetect, "SetProperty", args);
+                }
+
+                {
+                    QStringList propertiesMap;
+
+                    propertiesMap << "10.0.2.2";
+                    // TODO:XXX: CHANGE THIS!!!
+                    propertiesMap << "8.8.4.4";
+                    propertiesMap << "8.8.8.8";
+
+                    QVariantList args;
+                    args << "Nameservers.Configuration" << QVariant::fromValue(QDBusVariant(propertiesMap));
+                    service.callWithArgumentList(QDBus::AutoDetect, "SetProperty", args);
+                }
+            }
+        }
+    }
+}
+
+void TAP::mtuChanged(int mtu) {
+    struct ifreq ifr = {};
+    auto sock = socket(PF_INET, SOCK_DGRAM, 0);
+    if (sock < 0) {
+        qCritical() << "Failed to create a socket";
+        exit(1);
+    }
+    strncpy(ifr.ifr_name, mIfaceName.toLocal8Bit().data(), sizeof(ifr.ifr_name));
+    ifr.ifr_mtu = mtu - MY_ETH_TAP_HARD_FRAME_SIZE;
+    if (ioctl(sock, SIOCSIFMTU, &ifr) < 0) {
+        qCritical() << "Failed to set the MTU";
+        exit(1);
+    }
+    close(sock);
+
+    qDebug() << "MTU reset to" << mtu;
```

Comment:
Is a space after to necessary ? I never remember if this qDebug syntax does it automagically or not